demo/src/main/java/com/braintreepayments/api/demo/ComposeActivity.kt

```diff
@@ -0,0 +1,138 @@
+package com.braintreepayments.api.demo
+
+import android.net.Uri
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.BrowserSwitchPendingRequest
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.demo.utils.PendingRequestUtil
+import com.braintreepayments.api.demo.viewmodel.BrowserSwitchViewModel
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.Exception
+
+class ComposeActivity : ComponentActivity() {
+
+    private val viewModel by viewModels<BrowserSwitchViewModel>()
+
+    private lateinit var browserSwitchClient: BrowserSwitchClient
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        browserSwitchClient = BrowserSwitchClient()
+
+        setContent {
+            Column (modifier = Modifier.padding(10.dp)) {
+                BrowserSwitchButton {
+                    startBrowserSwitch()
+                }
+                BrowserSwitchResult(viewModel = viewModel)
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        PendingRequestUtil.getPendingRequest(this)?.let { startedRequest ->
+            val browserSwitchResult = browserSwitchClient.parseResult(startedRequest, intent)
+            browserSwitchResult?.let { result ->
+                viewModel.browserSwitchResult = result
+            } ?: run {
+                viewModel.browserSwitchError = Exception("User did not complete browser switch")
+            }
```

Comment:
random: I wonder if we could be more explicit in the SDK response here. Returning `null` from `parseResult()` feels like the SDK is implicitly communicating to the merchant that the state of browser switching is unknown.

Reply:
Yeah @scannillo brought that up in the [PayPal PR too](https://github.com/braintree/braintree_android/pull/883)... I'm just not sure we want to be opinionated about the response here.

Really all the method is doing is checking if the pending request passed in has a matching result - since we are now leaving the storage of requests up to merchants, we can't really be certain if whatever request they pass in is in fact still pending. We could return an empty `BrowserSwitchResult` with status `UNKNOWN`, but that seems like more work for the merchant to unwrap than just returning null, since we can't really assert still pending when we don't know if the request that was passed in is in fact still pending. What do you think?

Reply:
My gut feeling is by making it non-null, we eliminate some uncertainty for merchants by giving them a documented enum in the case of failure.

Both solutions (`null` and `enum` results) will require merchants to include branching logic (i.e. if/else, when, let/run) in their apps to handle success and failure scenarios separately.

Enums fit well with `when(result) {}` and `if/else` in Kotlin. I do wish Kotlin had an equivalent to `if/let` in Swift, I feel like that would make `null` feel better. But still, my concern is that `null` doesn't fail loudly enough.

Reply:
Agreed. I think it's outside the scope of this demo app PR, but created a ticket to update from `null` to a result with `UNKNOWN` status